### PR TITLE
Docker updates

### PR DESCRIPTION
**Pull request by ka7eh**
This PR makes the docker and docker-compose configs more modular, adds some wrappers for the main functionalities, and makes some small fixes. I didn't update the docs for now. I wanted to first see which parts of this changeset we should use. We can then update the docs in a separate PR.

### Main changes
- `docker-compose` variables are now handled through `.env` file, with some default values set in `docker-compose.yml`. To change the default values, you can create a `.env`file from `env-example` and edit this file. You can also specify the image tag to build or fetch from docker hub in `.env`.
- `run_docker.sh` is a wrapper to start the server with docker, which also handles some permission issues on mounted volumes. You can use it as is, or set `DEBUG` env var to 1 to mount the whole project folder into the container. This way you can edit and use your latest changes in the running docker container.
- `make_cases.sh` and `run_cases.sh` run `make_cases.py` and `run_cases.py` from `landsites_tools/simulation` via docker. You can pass any arguments that those python scripts accept to their bash script wrappers.
- There's now a Github Action that builds the docker image and add it to Github Container Registry under NorESMHub account.

### How you can test the changes:
- copy `docker/env-example` to `.env`: `cp ./docker/env-example .env`.
- build the docker image: `docker-compose -f docker-compose.dev.yml build`.
- Export `DEBUG` variable so all the scripts use `docker-compose.dev.yml`: `export DEBUG=1`.
- run the server: `./docker/run_docker.sh`.
- create the default task: `./docker/make_cases.sh`.
- run the default task: `./docker/run_cases.sh`.
- Create a new case by going to `localhost:5006/make_settings` and calling it `test`.
- create the test case: `./docker/make_cases.sh -f test.txt`.
- run the test case: `./docker/run_cases.sh -f landsites_tools/custom_settings/test.txt`.
- Go to `localhost:8888`, adjust the path in `notebooks/plot_example.ipynb` for the output of one of the cases you ran and run the notebook.

### CODE REVIEW

File: .machine/NREC_VM_setup/privileged/requirements_pip.txt [hunk 1]
```diff
@@ -15,3 +15,4 @@ cfunits==3.3.3
 pysqlite3
 jupyter
 bokeh
+dask[complete]
```
Comment (ka7eh) [inline .machine/NREC_VM_setup/privileged/requirements_pip.txt:18]: dask was needed for the example notebook, so I included it in the requirements.

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +0,0 @@
-numpy==1.19.5
```
Comment (ka7eh) [inline requirements.txt:1]: These are already being handled in `setup.py`.

**Review by evalieungh — COMMENTED** [review]
@MichalTorma and I looked at it this morning and think it looks good! Just need to test if everything works.

File: docker-compose.yml [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
+
 services:
   noresm_land_sites_platform:
-    image: "lassetk/noresm_land_sites_platform:latest"
-    mem_limit: 1024m # Max memory used
-    mem_reservation: 256M # Minimum memory used
-    cpus: 1.5 # Number of CPU cores to be used
+    build:
+      context: ./
+    image: "lassetk/noresm_land_sites_platform:${VERSION:-latest}"
```
Comment (huitang-earth) [inline docker-compose.yml:7]: Let's wait for the new organization to be set, and change this to the new address.

File: run_cases.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#! /usr/bin/bash
+
+docker-compose run -e USER=user noresm_land_sites_platform python landsites_tools/simulation/run_cases.py $@
```
Comment (huitang-earth) [inline run_cases.sh:3]: This file should be put in a specific folder?

File: run_docker.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+DOCKER_COMPOSE_FILE=docker-compose.yml
+if [[ $DEBUG == 1 ]]; then
+    DOCKER_COMPOSE_FILE=docker-compose.dev.yml
+fi
+
+echo "Using $DOCKER_COMPOSE_FILE"
+
+docker-compose -f ${DOCKER_COMPOSE_FILE} run --user root noresm_land_sites_platform chown -R user:user /home/user/NorESM_LandSites_Platform/data /home/user/NorESM_LandSites_Platform/landsites_tools/custom_settings /home/user/NorESM_LandSites_Platform/notebooks
+docker-compose -f $DOCKER_COMPOSE_FILE up
```
Comment (huitang-earth) [inline run_docker.sh:11]: Put this file into a specific folder?

**Review by huitang-earth — APPROVED** [review]

File: make_cases.sh [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#! /usr/bin/bash
+
+docker-compose run -e USER=user noresm_land_sites_platform python landsites_tools/simulation/make_cases.py $@
```
[the remark this answers is not in the document]
Comment (ka7eh) [reply make_cases.sh:3]: I like this idea of organizing docker-related scripts in one place. I'll move these to a docker folder.

**Pull request merged**